data/src/message.rs

```diff
@@ -107,7 +105,7 @@ impl From<Encoded> for proto::Message {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
My understanding is the only breakage is we renamed `Query.nick` to `Query.query`. We can use `#[serde(alias)]` to provide upgrade compatability from < 2025.1 to >= 2025.1

Reply:
Shit, `message::Source::Action` is also breaking as we added a payload to it :/

Reply:
We may want to do a bigger, more comprehensive update that can accomplish the following:

- Safely upgrades from 2025.1
- Is more easily upgradable in the future without breakage (no unit variants, only struct variants so  we can always add fields)

We need to separate the serialization format from the data format we use at runtime. Having them be the same makes it way too easy for breakage to occur. Having them separate and a simple `From / Into` conversion makes it super easy to see what needs to change in a _compatible_ way w/ the serialization format, when the data format changes.